src/pkg/fault/fault.go

```diff
@@ -379,68 +379,21 @@ func (pec printableErrCore) Values() []string {
 // Local aggregator
 // ---------------------------------------------------------------------------
 
-// Local constructs a new local bus to handle error aggregation in a
-// constrained scope.  Local busses shouldn't be passed down  to other
-// funcs, and the function that spawned the local bus should always
-// return `local.Failure()` to ensure that hard failures are propagated
-// back upstream.
-func (e *Bus) Local() *localBus {
-	return &localBus{
-		mu:  &sync.Mutex{},
-		bus: e,
-	}
-}
-
-type localBus struct {
-	mu      *sync.Mutex
-	bus     *Bus
-	current error
-}
-
-func (e *localBus) AddRecoverable(ctx context.Context, err error) {
-	if err == nil {
-		return
-	}
-
-	e.mu.Lock()
-	defer e.mu.Unlock()
-
-	if e.current == nil && e.bus.failFast {
-		e.current = err
+// Local constructs a new bus with a local reference to handle error aggregation
+// in a constrained scope.  This allows the caller to review recoverable errors and
+// failures within only the current codespace, as opposed to the global set of errors.
+// The function that spawned the local bus should always return `bus.Failure()` to
+// ensure that hard failures are propagated back upstream.
+func (e *Bus) Local() *Bus {
+	parent := e
+
+	// always use the root bus reference, if e is not the root.
+	if e.parent != nil {
+		e = e.parent
 	}
 
-	e.bus.logAndAddRecoverable(ctx, err, 1)
-}
-
-// AddSkip appends a record of a Skipped item to the local bus.
-// Importantly, skipped items are not the same as recoverable
-// errors.  An item should only be skipped under the following
-// conditions.  All other cases should be handled as errors.
-// 1. The conditions for skipping the item are well-known and
-// well-documented.  End users need to be able to understand
-// both the conditions and identifications of skips.
-// 2. Skipping avoids a permanent and consistent failure.  If
-// the underlying reason is transient or otherwise recoverable,
-// the item should not be skipped.
-func (e *localBus) AddSkip(ctx context.Context, s *Skipped) {
-	if s == nil {
-		return
+	return &Bus{
+		mu:     &sync.Mutex{},
+		parent: parent,
```

Comment:
this is still assigning the original value of `e` as the parent, not the updated one. It would be better to just have `parent: e.parent` here I think